userspace/libsinsp/plugin.cpp

```diff
@@ -1175,3 +1175,72 @@ bool sinsp_plugin::set_async_event_handler(async_event_handler_t handler) {
 
 	return rc == SS_PLUGIN_SUCCESS;
 }
+
+bool sinsp_plugin::dump(sinsp_dumper& dumper) {
+	if(!m_inited) {
+		throw sinsp_exception(std::string(s_not_init_err) + ": " + m_name);
+	}
+
+	if(!(m_caps & CAP_ASYNC)) {
+		throw sinsp_exception("plugin " + m_name + "without async events cap used as dumper");
+	}
+
+	if(!m_handle->api.dump) {
+		return false;
+	}
+
+	m_async_dump_handler = [&dumper](auto e) { dumper.dump(e.get()); };
```

Comment:
I assume we don't have thread safety issue here compared to the async event handler, right?

Reply:
Exactly!